packages/gatsby/cache-dir/loader.js

```diff
@@ -352,6 +319,22 @@ const queue = {
 
       if (!page) {
         console.log(`A page wasn't found for "${path}"`)
+
+        // Try to load the page directly.
+        //
+        // Usually the page either doesn't exist, resulting in a 404 error, or
+        // isn't available offline. Appending the query ensures the correct error
+        // message is displayed, or if the page is available online, it will
+        // load properly.
+
+        if (!window.location.search.match(/(\?|&)no-cache=1$/)) {
+          if (window.location.search) {
+            window.location.search += `&no-cache=1`
+          } else {
+            window.location.search = `?no-cache=1`
```

Comment:
We use `getResourcesForPathname` not only for current path (for example it's used in `PageRenderer` that is used to render page in background in some views (gatsbygram, showcase on gatsbyjs.org) and thisi might cause weird issues potentially - should this be moved to navigation code if we can't get resources?

Reply:
also used in `hovering` action - https://github.com/gatsbyjs/gatsby/blob/3fd71b036343c451a3cf4901b4e5c991fd154451/packages/gatsby/cache-dir/loader.js#L196 - it would be not great UX if we hover link for path we can't get resources and browser refreshes automatically?

Reply:
That's true - I actually noticed this slightly earlier and fixed it in my local code, but I haven't pushed it yet since I'm still trying to get everything to work together properly and fix the last few bugs.